### PR TITLE
Automate tests run in CI

### PR DESCRIPTION
**Pull request by alexandrebodin**
#### Description:

Automate tests runs with travis CI

- Runs unit tests (runs `npm run test` in every package)
- Runs end to end test against pg/mysql/sqlite/mongo databases (runs tests in files finishing by `test.e2e.js`)
- Runs **cypress** tests 

#### My PR is a:
- [ ] 💥 Breaking change
- [ ] 🐛 Bug fix #issueNumber
- [X] 💅 Enhancement
- [ ] 🚀 New feature

#### Main update on the:
- [ ] Admin
- [ ] Documentation
- [X] Framework
- [ ] Plugin

#### Manual testing done on the following databases:
- [X] MongoDB
- [x] MySQL
- [X] Postgres
- [x] SQLite

#### TODO

- [ ] Update README.md
- [x] Run against mysql and sqlite
- [x] Add snyk tests
- [x] Cleanup test scripts

### CODE REVIEW

File: packages/strapi-hook-bookshelf/lib/utils/connectivity.js [hunk 1]
```diff
@@ -76,6 +76,7 @@ module.exports = (scope, success, error) => {
     });
   })
     .catch((err) => {
+      console.log(err)
```
Comment (Aurelsicoko) [inline packages/strapi-hook-bookshelf/lib/utils/connectivity.js]: There is a `debug` option to check if we need to display the full log or not. We should remove this one, right?

Comment (alexandrebodin) [reply packages/strapi-hook-bookshelf/lib/utils/connectivity.js]: Updated

File: .travis.yml [hunk 2]
```diff
@@ -1,34 +1,77 @@
...
+        - psql -c 'create database strapi_test;' -U postgres
+      env:
+        - DB_STRING='--dbclient=postgres --dbhost=localhost --dbport=5432 --dbname=strapi_test --dbusername=postgres --dbpassword='
+
+    - <<: *e2e_tests
+      name: 'E2E Mysql'
+      before_install:
+        - mysql -e 'CREATE DATABASE strapi_test;'
+      env:
+        # mysql
```
Comment (lauriejim) [inline .travis.yml]: Do we have to keep this comments.

If yes - we have to add on for Postgres.

File: jest.config.e2e.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+module.exports = {
+  name: 'setup',
+  displayName: 'Setup',
+  testMatch: ['**/test/?(*.)+(spec|test).e2e.js'],
+  testEnvironment: 'node',
+  // globalSetup: '<rootDir>/test/globalSetup.js',
```
Comment (lauriejim) [inline jest.config.e2e.js]: Do we need to keep this line ?

File: packages/strapi-generate-api/test/endpoint.test.e2e.js [hunk 4]
```diff
@@ -679,180 +579,142 @@ describe('Test oneToOne relation (article - reference) with Content Manager', ()
...
+      await rq({
         url: `/tags/${tagToCreate.id}`,
         method: 'DELETE',
-        json: true
       });
 
-      const referenceToGet = await rq({
-        url: `/references/${referenceToCreate.id}`,
+      const { body: referenceToGet } = await rq({
+        url: `/references/${referenceToCreate.id || referenceToCreate._id}`,
```
Comment (lauriejim) [inline packages/strapi-generate-api/test/endpoint.test.e2e.js]: I suppose you add it to manager mongoose entries.
But `.id` also exist for mongoose nop ?

File: packages/strapi/bin/strapi-start.js [hunk 5]
```diff
@@ -11,13 +11,59 @@ const path = require('path');
...
+      strapi.reload.isReloading = true;
+      strapi.reload();
+    }
+  };
+
+  const watcher = chokidar.watch(appPath, {
+    ignoreInitial: true,
+    ignored: [
+      /(^|[/\\])\../,
+      '**/tmp',
```
Comment (lauriejim) [inline packages/strapi/bin/strapi-start.js]: It's `.tmp` or `tmp` ?

Comment (derrickmehaffy) [reply packages/strapi/bin/strapi-start.js]: That would likely depend on linux/windows. probably best to handle both cases for the future

File: packages/strapi-generate-api/test/endpoint.test.e2e.js (hunk 4 above)
Comment (alexandrebodin) [reply packages/strapi-generate-api/test/endpoint.test.e2e.js]: Forgot to remove it

**Pull request merged**